Add a post announcing our CDN on Community (#81)
readthedocs/blog: announcing-community-cdn.rst

> Shipping a CDN on Read the Docs Community

Reviewer: Meta tags are missing here (`.. post` with author's name and date, and `.. meta` with SEO info)

How much faster likely depends on how far away you live from Virginia,
which is where our servers have traditionally lived.

We have recently enabled a CDN on **all Read the Docs Community sites**,
generously sponsored by `CloudFlare`_.
This post will talk a bit more about how we implemented this,
and why we're excited about it.

Hosting thousands of domains is hard
------------------------------------

Traditionally the largest problem that we've had with rolling out features to all of our documentation sites is scale.
We host thousands of custom domains for our users,
and any solution needs to work across all of them.
This has presented a number of issues in the past,
but CDN is one of the most complicated.

To make a CDN function across all our sites,
every data center that the CDN has needs to work with all our custom domains.
Imagine we have 5,000 domains and there are 100 data centers across the world,
this means that ``5,000 * 100 = 500,000`` different endpoints need to be configured for this to work at scale.

We are lucky that CloudFlare has the global scale to be able to donate us this service.
Specifically,

> their `SSL for SAAS`_ service is what we're using for both SSL and CDN across all our custom domains

Reviewer: ```suggestion
their `SSL for SAAS`_ service is what we're using for both SSL and CDN across all our custom domains.
```

This allows us to offload the complexity to CloudFlare,
and only focus on our integration.

Implementation
--------------

One of the coolest things that CloudFlare's CDN offers is something called `Cache Tags`_.
This lets us add a ``Cache-Tag`` header to all the documentation that we serve,
and bust the cache using just that tag.

An example,
when you load our docs,
we will return a header::

    GET https://docs.readthedocs.io/en/latest/
    ...

>     Cache_tag: docs-latest,docs

Reviewer: ```suggestion
    Cache-Tag: docs-latest,docs
```

Just consistency with the mention in the previous paragraph.

Reviewer: ```suggestion
    Cache-Tag: docs,docs-latest
```

Sort the values in the same way they are mentioned in the following paragraph.

We return a tag that matches both the ``$project``, and the ``$project-$version``.

> This allows us to bust the cache for a specific version,

Reviewer: ```suggestion
This allows us to invalidate the cache for a specific version,
```

Invalidate sounds more familiar to me when talking about cache.

Reviewer: Hrm... There are more uses of the "bust" word in the next paragraphs. I assume it's just to not repeat yourself too much.

or across the entire project.

As you know,
cache invalidation is one of the harder problems,
so we take a pretty conservative approach.
We bust the cache in the following scenarios:

* Bust Project cache on Project or Domain save
* Bust Version cache on documentation builds for specific versions

The client code is quite simple,

> a single HTTP request with a list of ``tags`` to clear.

Reviewer: ```suggestion
a single HTTP request to Cloudflare's API with a list of ``tags`` to clear.
```

.. post:: May 12, 2020
   :tags: community, cdn, hosting
   :author: Eric

.. meta::
   :description lang=en:

>       Read the Docs Community now has a CDN enabled for all projects.

Reviewer 2: This meta description is probably a little too short and probably just won't get used by search engines. It's a pretty [inexact science](https://moz.com/learn/seo/meta-description) whether one gets used but I'd suggest something like this:

```suggestion
      Read the Docs Community now has Cloudflare CDN enabled for all projects resulting in faster and more secure documentation worldwide.
```
